.github/workflows/deploy_artifacts.yaml

```diff
@@ -0,0 +1,111 @@
+# Deploys artifacts using maven and optionally deploys an image to quay
+name: Reusable artifacts deploy
+
+on:
+  workflow_call:
+    inputs: 
+      createDockerImage:
+        description: 'Whether to create a Docker image, caller must have Dockerfile in the repository'
+        required: false
+        type: boolean
+        default: false
+      quayRepository:
+        description: The quay repository to upload the image to. The repository must belong to the dockstore quay organization.
+        required: false
+        type: string
+      dockerContext:
+        description: The Docker context containing the Dockerfile of the image to build and push.
+        required: false
+        default: .
+        type: string
+
+env:
+  IS_DEVELOP_SNAPSHOT: ${{ github.ref_type == 'branch' && github.ref_name == 'develop' }}
+
+jobs:
+  deploy_maven:
+    name: Maven deploy ${{ github.ref_type == 'tag' && 'tagged' || 'snapshot' }} release
+    runs-on: ubuntu-22.04
+
+    steps:
+      - uses: actions/checkout@v4
+      # Step that does that actual cache save and restore
+      - uses: actions/cache@v4
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+           ${{ runner.os }}-maven-
+
+      - name: install git secrets
+        run: |
+          wget --no-verbose -O git-secrets-1.3.0.tar.gz https://github.com/awslabs/git-secrets/archive/1.3.0.tar.gz
+          tar -zxf git-secrets-1.3.0.tar.gz
+          cd git-secrets-1.3.0
+          sudo make install   
+
+      - name: Set up JDK
+        uses: actions/setup-java@v4
+        with:
+          java-version: '21.0.2+13.0.LTS'
+          distribution: 'adopt'
+          # settings.xml configuration
+          server-id: ${{ github.ref_type == 'tag' && 'central' || 'snapshots' }}
+          server-username: DEPLOY_USERNAME
+          server-password: DEPLOY_TOKEN
+
+      - name: Set changelist version
+        run: |
+          set -x
+          if ${{ github.ref_type == 'tag' }}; then
+            # Check that the tag follows semantic versioning. Note that the regex is not super strict
+            if [[ ${{ github.ref_name }} =~ ^[0-9]+\.[0-9]+\.[0-9a-zA-Z-]+$ ]]; then
```

Comment:
This is a simple check just to make sure that the tag generally follows semantic versioning. There is a much stricter regex out there, but I figured it might be overkill here